runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsService.java

```diff
@@ -70,22 +70,26 @@ public void forEachPlayer(final BiConsumer<Player, Color> consumer)
 					consumer.accept(player, config.getOwnNameColor());
 				}
 			}
-			else if (config.drawFriendNames() && player.isFriend())
+			if (!config.onlyPlayerMenu())
```

Comment:
This logic should probably also be placed near whatever draws player names, and not here which just allows parts of the plugin to pass in a consumer.